packages/worker/src/contract/systemContract.service.ts

```diff
@@ -26,7 +26,7 @@ export class SystemContractService {
         const bytecode = await this.blockchainService.getCode(contract.address);
         // some contract might not exist on the environment yet
         if (bytecode !== "0x") {
-          await this.addressRepository.add({
+          await this.addressRepository.upsert({
```

Comment:
@Romsters in the case you've described it's unlikely that the bytecodes will be different, but still I'd rather wrap `add` with try / catch or use `add` with `on conflict do nothing` not to overwrite the bytecode with an obsolete value.

Reply:
It will not update the bytecode in that case as it has block number condition that checks if the data is not outdated before updating